homeassistant/components/sensor/hive.py

```diff
@@ -0,0 +1,486 @@
+"""
+Support for the Hive devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/hive/
+"""
+import logging
+from datetime import datetime
+
+from homeassistant.components.climate import (STATE_AUTO, STATE_HEAT,
+                                              STATE_OFF, STATE_ON)
+from homeassistant.components.hive import DATA_HIVE
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.helpers.entity import Entity
+
+DEPENDENCIES = ['hive']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, hivedevice, discovery_info=None):
+    """Set up Hive sensor devices."""
+    session = hass.data.get(DATA_HIVE)
+
+    add_devices([HiveSensorEntity(hass, session, hivedevice)])
+
+
+class HiveSensorEntity(Entity):
+    """Hive Sensor Entity."""
+
+    def __init__(self, hass, Session, HiveDevice):
+        """Initialize the sensor."""
+        self.node_id = HiveDevice["Hive_NodeID"]
+        self.node_name = HiveDevice["Hive_NodeName"]
+        self.device_type = HiveDevice["HA_DeviceType"]
+        self.node_device_type = HiveDevice["Hive_DeviceType"]
+        self.hass = hass
+        self.session = Session
+        self.session.sensor = self.session.core.Sensor()
+        if self.device_type == "Hive_Device_BatteryLevel":
+            self.batt_lvl = None
+
+        self.session.entities.append(self)
+
+    def handle_update(self, updatesource):
+        """Handle the new update request."""
+        if self.device_type + "." + self.node_id not in updatesource:
+            self.schedule_update_ha_state()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        friendly_name = "Sensor"
+
+        if self.device_type == "Heating_CurrentTemperature":

```

Comment:
Use a dictionary to map the names to the device types, and lookup the the dictionary here.

Reply:
done